### PR TITLE
Style/DateTime tweaks

### PR DESCRIPTION
**Pull request by deivid-rodriguez**
As per the discussion in #6189, I'm proposing my alternative. Even if closed, it could still be useful for people who want to keep this code as a custom cop, so I'm opening the PR anways :) 

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [ ] Added tests.
* [x] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Run `rake default` or `rake parallel`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: http://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by deivid-rodriguez** [comment]
CI is green now after a rebase.

File: CHANGELOG.md [hunk 1]
```diff
@@ -40,6 +40,7 @@
 * [#4832](https://github.com/rubocop-hq/rubocop/issues/4832): Change the path pattern (`*`) to match the hidden file. ([@koic][])
 * `Style/For` now highlights the entire statement rather than just the keyword. ([@rrosenblum][])
 * Disable `Performance/CaseWhenSplat` and its auto-correction by default. ([@rrosenblum][])
+* [\#6199](https://github.com/rubocop-hq/rubocop/pull/6199): `Style/DateTime` has been moved to disabled by default. ([@deivid-rodriguez][])
```
Comment (bbatsov) [inline CHANGELOG.md]: You've got an extra `\` here.

**Comment by bbatsov** [comment]
I think you can also expand a bit the rationale section of the cop and also add there was this is disabled by default.

**Comment by deivid-rodriguez** [comment]
@bbatsov I tried to improve the docs, not sure if it's what you had in mind.

**Comment by mvz** [comment]
This cop is enabled again in 0.60.0.

**Comment by deivid-rodriguez** [comment]
Yup, sounds like a bad rebase of https://github.com/rubocop-hq/rubocop/pull/6293.

**Comment by AlexWayfer** [comment]
> Yup, sounds like a bad rebase of #6293.

I'm trying to resolve this in #6451.

**Pull request merged**